src/Common/Filesystem.h

```diff
@@ -19,13 +19,22 @@
 
 #pragma once
 
+#include "Config.h"
+
 #ifdef _MSC_VER
 #include <filesystem>
 namespace filesystem = std::experimental::filesystem;
-#elif defined(__APPLE__)
-#include <boost/filesystem>
-namespace filesystem = boost::filesystem;
 #else
+#if __has_include(<filesystem>)
 #include <filesystem>
 namespace filesystem = std::filesystem;
+#elif __has_include(<experimental/filesystem>)
+#include <experimental/filesystem>
+namespace filesystem = std::experimental::filesystem;
+#elif __has_include(<boost/filesystem.hpp>)
+#include <boost/filesystem.hpp>
```

Comment:
Note that boost/filesystem requires compilation. Avoid introducing unnecesssary dependencies especially heavy-weight libraries like boost if possible.